src/odemis/gui/comp/overlay/world.py

```diff
@@ -1148,16 +1076,171 @@ def _points_to_rect(left, top, right, bottom):
         return left, top, right - left, bottom - top
 
     def _update_view_positions(self, b_start_pos, b_end_pos):
-        """Given that the canvas has been shifted or rescaled, update the view positions of the ruler"""
-        logging.debug("Updating view position of ruler %s", self)
+        """ Given that the canvas has been shifted or rescaled, update the view positions of the line """
+        logging.debug("Updating view position of line %s", self)
         self.v_start_pos = Vec(self.cnvs.buffer_to_view(b_start_pos))
         self.v_end_pos = Vec(self.cnvs.buffer_to_view(b_end_pos))
-        self._calc_edges()
+        self.calc_edges()
+
+    @abstractmethod
+    def draw(self, ctx, selected, shift=(0, 0), scale=1.0, canvas=None, font_size=None):
+        pass
+
+
+NONE_RULER_MODE = 0
+MOVE_RULER_MODE = 1
+EDIT_START_RULER_MODE = 2
+EDIT_END_RULER_MODE = 3
+
+
+class Ruler(GenericGadgetLine):
```

Comment:
Let's call this class `RulerGadget` and `LabelTool` -> `LabelGadget`. This way, everything connected to the `GadgetOverlay` has the same keyword.